app/src/main/java/com/example/dictionary/di/DataSourceQualifiers.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.dictionary.di
+
+import javax.inject.Qualifier
+
+@Qualifier
```

Comment:
nice job, this is indeed what it should look like but as a matter of preferences, if you don't like it you can just  create a contract for LocalDS and one for RemoteDs, like this : LocalDataSourceImpl : LocalDataSource, RemoteDataSourceImpl : RemoteDataSource and the repository would look like : Repository (localDs : LocalDataSource, remoteDs : RemoteDataSource). But as I've said, it's a matter of preference

Reply:
It's more agnostic with one common interface and 2 different implementations and that's why I chose this way